tests/integration/sync_test.go

```diff
@@ -25,82 +137,443 @@ func getKongAddress() string {
+	}
+
+	tests := []struct {
+		name          string
+		kongFile      string
+		expectedState utils.KongRawState
+	}{
+		{
+			name:     "creates a service",
+			kongFile: "testdata/sync/001-create-a-service/kong.yaml",
+			expectedState: utils.KongRawState{
+				Services: svc1,
+			},
+		},
+		{
+			name:     "create services and routes",
+			kongFile: "testdata/sync/002-create-services-and-routes/kong.yaml",
+			expectedState: utils.KongRawState{
+				Services: svc1,
+				Routes:   route1_205_214,
+			},
+		},
+	}
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
-			if err := cleanUpEnv(t); err != nil {
-				t.Errorf(err.Error())
-			}
-
-			deckCmd.SetArgs([]string{"sync", "-s", tc.kongFile})
-			if err := deckCmd.Execute(); err != nil {
-				t.Errorf(err.Error())
-			}
-			testKongState(t, client, tc.expectedServices)
+			runWhenKong(t, client, ">=2.0.5")
+			runWhenKong(t, client, "<=2.1.4")
+			teardown := setup(t)
+			defer teardown(t)
+
+			sync(tc.kongFile)
+			testKongState(t, client, tc.expectedState, nil)
 		})
 	}
 }
+
+func Test_Sync_ServicesRoutes_From_2_2_2_to_2_6_0(t *testing.T) {
+	// setup stage
+	client, err := getTestClient()
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+
+	tests := []struct {
+		name          string
+		kongFile      string
+		expectedState utils.KongRawState
+	}{
+		{
+			name:     "creates a service",
+			kongFile: "testdata/sync/001-create-a-service/kong.yaml",
+			expectedState: utils.KongRawState{
+				Services: svc1,
+			},
+		},
+		{
+			name:     "create services and routes",
+			kongFile: "testdata/sync/002-create-services-and-routes/kong.yaml",
+			expectedState: utils.KongRawState{
+				Services: svc1,
+				Routes:   route1_20x,
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			runWhenKong(t, client, ">=2.2.2")
+			runWhenKong(t, client, "<=2.6.0")
+			teardown := setup(t)
+			defer teardown(t)
+
+			sync(tc.kongFile)
+			testKongState(t, client, tc.expectedState, nil)
+		})
+	}
+}
+
+func Test_Sync_ServicesRoutes_2_7_0(t *testing.T) {
+	// setup stage
+	client, err := getTestClient()
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+
+	tests := []struct {
+		name          string
+		kongFile      string
+		expectedState utils.KongRawState
+	}{
+		{
+			name:     "creates a service",
+			kongFile: "testdata/sync/001-create-a-service/kong.yaml",
+			expectedState: utils.KongRawState{
+				Services: svc1_207,
+			},
+		},
+		{
+			name:     "create services and routes",
+			kongFile: "testdata/sync/002-create-services-and-routes/kong.yaml",
+			expectedState: utils.KongRawState{
+				Services: svc1_207,
+				Routes:   route1_20x,
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			runWhenKong(t, client, "==2.7.0")
+			teardown := setup(t)
+			defer teardown(t)
+
+			sync(tc.kongFile)
+			testKongState(t, client, tc.expectedState, nil)
+		})
+	}
+}
+
+// func Test_Sync_Plugins(t *testing.T) {
```

Comment:
Either remove this code or get it working. Commented out code with VCS is a no no.

Reply:
sure, just wanted to have a +1 on the overall structure...I'll add them back now